crates/storage/provider/src/transaction.rs

```diff
@@ -1,21 +1,18 @@
 #![allow(dead_code)]
-use std::{
-    fmt::Debug,
-    ops::{Deref, DerefMut},
-};
-
 use reth_db::{
```

Comment:
imo moving this from `stages` to `provider` is only a small improvement.

imo there should be a new create for db related types.

Reply:
ok, so at least the `Transaction` type would be moved, but I'm not sure what else, the separation would look kind of like this:
 * `reth_provider`: high level abstractions for using the db, consumed by RPC, the executor, engine api, etc
 * `reth_???`: lower-level abstractions (such as `Transaction`) for using the db, but not the lowest, used to implement providers and stages
 * `reth_db`: lowest level primitivies, defines tables, codecs, and low level abstractions for working with the database, managing transactions, cursors, etc